### ISSUE TITLE
Clean up decision task hacks

### ISSUE THREAD

**Issue by imbstack** [taskcluster/taskcluster#2924]
Things noted by @tomprince:

1. `TransformSequence` in `taskcluster/src/transforms/__init__.py` is not necessary
2. We should also change `basestring` to `six.text_type` wherever used
3. Consider using level-1 vs level-3 to keep decision task stuff on its own workers instead of a `decision` worker.

**Comment by tomprince** [comment]
>     3\. Consider using level-1 vs level-3 to keep decision task stuff on its own workers instead of a `decision` worker.

More specifically to keep pull-request tasks on separate workers for tasks for pushes. This is probably most relevant for any tasks that are involved in the release process (including the decision task).

**Comment by tomprince** [comment]
Also, consider moving `tascluster/src` to taskcluster/taskcluster_taskgraph` or the like. The convention releng has adopted is `<something>_taskgraph` for local taskgraph code. (Though nothing enforces this or cares)

**Comment by 090max** [comment]
I want to work on this .

**Comment by imbstack** [comment]
Thank you, @090max! I've assigned you. All work takes place inside the `taskcluster` directory and `.taskcluster.yml` file. Let me know if you want any further pointers.

**Comment by 090max** [comment]
sure

**Comment by 090max** [comment]
Just had these two queries:
1) Can you please elaborate task 3 "Consider using level-1 vs level-3 to keep decision task stuff on its own workers instead of a decision worker."
2) By removing TransformSequence , all the functions proceeding @transforms.add would no longer be required in the file taskcluster/src/transforms/__init__.py , right ?

**Comment by imbstack** [comment]
>  Can you please elaborate task 3 "Consider using level-1 vs level-3 to keep decision task stuff on its own workers instead of a decision worker."

Ah sure. Right now we use a `release` worker to do certain kinds of tasks and `ci` to do others. These are defined in `taskcluster/ci/config.yml` and you can see where they're used elsewhere. We could instead use workers called `ci-level-1` and `ci-level-2` or something like that.

Now that I think about it, this is a slightly more involved change. Let's do the other changes first and then we can handle this in a followup once you're more familiar with this code.

>  By removing TransformSequence , all the functions proceeding @transforms.add would no longer be required in the file taskcluster/src/transforms/\_\_init\_\_.py , right ?

I believe that's correct, yeah.

**Comment by 5aumy4** [comment]
I am new to contributing to open source orgs
Can I work on this?

**Comment by abdelquodr** [comment]
Is this issue fixed?. because it is still among the unassigned task on codetribute.

**Comment by lotas** [comment]
@ahal I think you are the most relevant person to ask if this is still needed? 
Thanks

**Comment by ahal** [comment]
Yeah, a bit of a mish mash of things in here, but looks like it's mostly still relevant. We should be running the decision task with Python 3 at least now, so the `basestring` stuff should be fixed.
